### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by Aleksey-Terzi**
1) Improved load and shutdown process as try to prevent bug when not all links loaded on server startup

2) Added new command "cg reload" to reload gearblocks and links from the database

3) Introduced rule to check if activation is allowed - all blocks (gearblocks and bridge blocks) in construction should be either non-reinforced or reinforced by the same group. This needed to prevent "cheating" using CastleGates

4) Introduced rule for transfer power - if gearblock is reinforced by another group than start gearblock - then activation is not locked, but this gearblock just ignored and power not transfers here. This is needed to prevent "bridge-griefing" by placing gearblock near someones CastleGates' construction

### CODE REVIEW

**Comment by Aleksey-Terzi** [comment]
All functionality is tested on my local environment.

**Comment by ProgrammerDan** [comment]
compiled against api 1.12 -- tested to still work on 1.10.2?

**Comment by Aleksey-Terzi** [comment]
> compiled against api 1.12 -- tested to still work on 1.10.2?

Yes, this is correct.

File: src/main/java/com/aleksey/castlegates/plugins/citadel/Citadel.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public Integer getGroupId() {
+        return this.groupId;
+    }
+
+    public boolean canAccessDoors(Location location) {
+        if(!this.hasAccess) return false;
+
+        Reinforcement rein = vg.civcraft.mc.citadel.Citadel.getReinforcementManager().getReinforcement(location);
+
+        return rein == null || !(rein instanceof PlayerReinforcement)
```
Comment (ProgrammerDan) [inline src/main/java/com/aleksey/castlegates/plugins/citadel/Citadel.java]: you can have two groups be the same group but have different group IDs, due to merging. :-/

**Pull request merged**